### ISSUE TITLE
despite payload update with `{"message":"Accepted.","status":202}`, errored out release is not updated

### ISSUE THREAD

**Issue by dimpase** [zenodo/zenodo#2055]
We are trying to fix the Zenodo DOI integration for 
https://zenodo.org/account/settings/github/repository/sagemath/sage#
on https://github.com/sagemath/sage/issues/100 - 
but despite renewing the webhook and editing the release, and [getting new payload accepted](https://github.com/sagemath/sage/settings/hooks/253557611), it remains red, with the stale `"errors": "Unknown error occured."`
message.

Please have a look into it.

**Comment by Glignos** [comment]
Editing a release despite despatching a new event is not handled by our GitHub integration, since it will only try to publish release events.
Sometimes releases might fail due to unexpected issues e.g. network timeouts.
For the time being, we have improved our error handling to provide a better insight into what went wrong to our users, and in the future, we plan to work towards fail proofing the procedure.

P.S: I have now manually reprocessed the release and it is not correctly published.
